File: viewjava/app/src/main/java/pers/shawxingkwok/kdatastore/viewjava/app/MainActivity.java
package pers.shawxingkwok.kdatastore.viewjava.app;

import android.annotation.SuppressLint;
import androidx.annotation.OptIn;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatDelegate;
import pers.shawxingkwok.kdatastore.KDataStore;
import pers.shawxingkwok.kdatastore.viewjava.settings.Settings;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (savedInstanceState == null) {
            getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.container, MainFragment.newInstance())
                .commitNow();
        }

        // link stored data to the corresponding functionality via LiveData.
        Settings.getTheme().getLiveData().observe(this, (theme)->{
            switch (theme){
                case FOLLOW_SYSTEM:
                    AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
                    break;

                case DARK:
                    AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
                    break;

                case LIGHT:
                    AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
                    break;
            }

            getDelegate().applyDayNight();
        });
    }
}

File: demo/view/src/main/java/pers/apollokwok/prefstore/example/view/MainActivity.kt
package pers.apollokwok.prefstore.example.view

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import pers.apollokwok.prefstore.example.view.ui.main.MainFragment

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState == null) {
            supportFragmentManager
                .beginTransaction()
                .replace(R.id.container, MainFragment())
                .commitNow()
        }
    }
}